src/platform/activation/workspaceActivation.node.ts

```diff
@@ -1,40 +1,24 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-'use strict';
-
-import { inject, injectable, multiInject, optional } from 'inversify';
+import { inject, injectable } from 'inversify';
 import { TextDocument } from 'vscode';
+import { sendActivationTelemetry } from '../../telemetry/envFileTelemetry.node';
 import { IPythonExtensionChecker } from '../api/types';
-import { IActiveResourceService, IDocumentManager, IWorkspaceService } from '../common/application/types';
+import { IWorkspaceService, IActiveResourceService, IDocumentManager } from '../common/application/types';
 import { PYTHON_LANGUAGE } from '../common/constants';
 import { IFileSystem } from '../common/platform/types.node';
 import { IDisposable, Resource } from '../common/types';
 import { Deferred } from '../common/utils/async';
 import { IInterpreterService } from '../interpreter/contracts.node';
-import { sendActivationTelemetry } from '../../telemetry/envFileTelemetry.node';
-import {
-    IExtensionActivationManager,
-    IExtensionActivationService,
-    IExtensionSingleActivationService,
-    IExtensionSyncActivationService
-} from './types';
 import { traceDecoratorError } from '../logging';
+import { IExtensionSingleActivationService } from './types';
 
 @injectable()
-export class ExtensionActivationManager implements IExtensionActivationManager {
+export class WorkspaceActivation implements IExtensionSingleActivationService {
```

Comment:
This class split into two parts. 
- WorkspaceActivation - initializes getting interpreters and sending workspace telemetry
- ExtensionActivationManager - initializes the IExtensionSingleActivationServices (both sync and async)